static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -55,97 +111,127 @@ function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorP
       options: [
         {
-          value: '_everyone',
-          label: t('Everyone'),
-          disabled: dashboardCreator?.id !== currentUser.id,
+          value: '_allUsers',
+          label: t('All users'),
+          disabled: !isCurrentUserDashboardOwner,
         },
       ],
     },
-    // [WIP: Selective edit access to teams]
-    // {
-    //   value: '_teams',
-    //   label: t('Teams'),
-    //   options: teams.map(makeTeamOption),
-    //   showToggleAllButton: true,
-    //   disabled: true,
-    // },
+    {
+      value: '_teams',
+      label: t('Teams'),
+      options: teams.map(makeTeamOption),
+      showToggleAllButton: isCurrentUserDashboardOwner,
+      disabled: !isCurrentUserDashboardOwner,
+    },
   ];
 
-  // Handles state change when dropdown options are selected
-  const onSelectOptions = newSelectedOptions => {
-    const newSelectedValues = newSelectedOptions.map(
-      (option: {value: string}) => option.value
-    );
-    if (newSelectedValues.includes('_everyone')) {
-      setselectedOptions(['_everyone', '_creator', ...teamIds]);
-    } else if (!newSelectedValues.includes('_everyone')) {
-      setselectedOptions(['_creator']);
-    }
-  };
-
-  // Creates or modifies permissions object based on the options selected
-  function getDashboardPermissions() {
-    return {
-      isCreatorOnlyEditable: !selectedOptions.includes('_everyone'),
-    };
-  }
+  // Save and Cancel Buttons
+  const dropdownFooterButtons = (
+    <FilterButtons>
+      <Button
+        size="sm"
+        onClick={() => {
+          setMenuOpen(false);
+        }}
+        disabled={!isCurrentUserDashboardOwner}
+      >
+        {t('Cancel')}{' '}
+      </Button>
+      <Button
+        size="sm"
+        onClick={() => {
+          const isDefaultState =
+            !defined(dashboard.permissions) && selectedOptions.includes('_allUsers');
+          const newDashboardPermissions = getDashboardPermissions();
+          if (
+            !isDefaultState &&
+            !isEqual(newDashboardPermissions, dashboard.permissions)
+          ) {
+            onChangeEditAccess?.(newDashboardPermissions);
+          }
+          setMenuOpen(!isMenuOpen);
+        }}
+        priority="primary"
+        disabled={!isCurrentUserDashboardOwner || !hasUnsavedChanges}
+      >
+        {t('Save Changes')}
+      </Button>
+    </FilterButtons>
+  );
 
   const dropdownMenu = (
     <StyledCompactSelect
       size="sm"
       onChange={newSelectedOptions => {
         onSelectOptions(newSelectedOptions);
-      }}
-      onClose={() => {
-        const isDefaultState =
-          !defined(dashboard.permissions) && selectedOptions.includes('_everyone');
-        const newDashboardPermissions = getDashboardPermissions();
-        if (!isDefaultState && !isEqual(newDashboardPermissions, dashboard.permissions)) {
-          onChangeEditAccess?.(newDashboardPermissions);
-        }
+        setHasUnsavedChanges(true);
```

Comment:
Instead of storing this state, is there a way we can derive it from comparing the dashboard prop and the state? e.g. if I select a team and unselect it to return to the original state, it still shows me in an unsaved state when saving wouldn't really change anything.

I think this would be similar to the logic you have for checking when to fire the request. You could possibly extract that logic into a helper